LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/IotHubLogger.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("LoRaWan.Tests.Unit")]
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
+
+namespace Logger
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Text;
+    using System.Threading.Tasks;
+    using LoRaWan;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.DependencyInjection.Extensions;
+    using Microsoft.Extensions.Logging;
+
+    public record IotHubLoggerConfiguration(LogLevel LogLevel, EventId EventId, bool UseScopes);
+
+    internal sealed class IotHubLoggerProvider : ILoggerProvider
+    {
+        private readonly ConcurrentDictionary<string, IotHubLogger> loggers = new();
+        private readonly Lazy<Task<ModuleClient>> moduleClientFactory;
+
+        public IotHubLoggerConfiguration Configuration { get; }
+        public LoggerExternalScopeProvider? ScopeProvider { get; }
+
+        public IotHubLoggerProvider(IotHubLoggerConfiguration configuration)
```

Comment:
this setup won't support updates to the log level without re-starting the box. The ConsoleLogger has support for dynamic updates, I think we should do this here as well.

Reply:
Since we can't use that feature, yet, I created a story for it: https://github.com/Azure/iotedge-lorawan-starterkit/issues/989. Will prepare for the story and implement what you suggested

Reply:
Addressed this in 4250a35c1e15a4fdf7edbae7525cc842b1ba6c73